djangoProject3/DFS/serializer.py

```diff
@@ -0,0 +1,176 @@
+        return value
+
+
+
+class LoginSerializer(serializers.Serializer):
+    username = serializers.CharField()
+    password = serializers.CharField()
+
+    def validate(self, data):
+        user = authenticate(**data)
+        if user and user.is_active:
+            return user
+        raise serializers.ValidationError("Incorrect Credentials")
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ['id', 'username', 'email', 'first_name', 'last_name']
+
+
+class UserProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserProfile
+        fields = ['user', 'profile_picture', 'bio']
+
+
+class AuthorSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Author
+        fields = ['id', 'name']
+
+
+class GenreSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Genre
+        fields = ['id', 'name']
+
+
+class BookSerializer(serializers.ModelSerializer):
+    author = serializers.CharField(source='author.name')
+    genres = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Book
+        fields = ['id', 'title', 'author', 'genres', 'price', 'price_id']
+
+    def get_genres(self, obj):
+        return [genre.name for genre in obj.genres.all()]
+
+
+class ReviewSerializer(serializers.ModelSerializer):
+    user = serializers.CharField(source='user.username')
+    book = serializers.CharField(source='book.title')
+
+    class Meta:
+        model = Review
+        fields = ['id', 'user', 'book', 'text', 'created_at']
+
+    def update(self, instance, validated_data):
+        instance.text = validated_data.get('text', instance.text)
+        instance.save()
+        return instance
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+        book_tit = validated_data['book']
+        text = validated_data['text']
+
+        book = Book.objects.get(title=book_tit['title'])
+
+        review = Review.objects.create(user=user, book=book, text=text)
+        return review
+
+
+class RatingSerializer(serializers.ModelSerializer):
+    user = serializers.CharField(source='user.username')
+    book = serializers.CharField(source='book.title')
+
+    class Meta:
+        model = Rating
+        fields = ['id', 'user', 'book', 'rating']
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+        book_tit = validated_data['book']
+        rating_value = validated_data['rating']
+
+        book = Book.objects.get(title=book_tit['title'])
+
+        existing_rating = Rating.objects.filter(user=user, book=book).first()
+        if existing_rating:
+            existing_rating.rating = rating_value
+            existing_rating.save()
+            return existing_rating
+
+        rating = Rating.objects.create(user=user, book=book, rating=rating_value)
+        return rating
+
+
+class NotificationSerializer(serializers.ModelSerializer):
```

Comment:
everything is in one file, everything should be i its file
i.e notifications.py, order.py, user.py